generator.cpp

```diff
@@ -13,17 +13,17 @@ std::unordered_set<uint64_t> visited;
  * Explore and print all possible position under a given depth.
  * symetric positions are printed only once.
  */
-void explore(const Position &P, char* pos_str, const int depth) {
+void explore(const Position &P, char* const pos_str, const int depth) {
   uint64_t key = P.key3();
   if(visited.count(key)) return;  // already explored position
   visited.insert(key);            // flag new position as visited
 
-  int nb_moves = P.nbMoves();
+  const int nb_moves = P.nbMoves();
   if(nb_moves <= depth)
   std::cout << pos_str << std::endl;
   if(nb_moves >= depth) return;  // do not explore at further depth
 
-  for(int i = 0; i < Position::WIDTH; i++) // explore all possible moves
+  for(int i = 0; i < Position::WIDTH; ++i) // explore all possible moves
```

Comment:
Always use pre-increment when at all possible: post-increment instructs the processor to store a temporary variable in the register that doesn't need to exist there.

Reply:
for simple type as int, there is not actual performance difference between pre-increment and post increment as in C. There is a actual difference for C++ classes.

And for a totally suggestive point of view, I feel the post-increment notation more comprehensible when reading code.

Reply:
Fair